File: src/main/java/com/qlik/demo/java/TrainRepository.java
package com.qlik.demo.java;

import org.springframework.data.mongodb.core.MongoTemplate;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
public class TrainRepository {

    private final MongoTemplate mongoTemplate;

    public TrainRepository(MongoTemplate mongoTemplate) {
        this.mongoTemplate = mongoTemplate;
    }

    public List<TrainDocument> findAll() {
        return mongoTemplate.findAll(TrainDocument.class);
    }
}

File: src/main/kotlin/com/qlik/demo/kotlin/TrainRepository.kt
package com.qlik.demo.kotlin

import com.qlik.demo.kotlin.model.TrainDocument
import org.springframework.data.mongodb.core.MongoTemplate
import org.springframework.stereotype.Repository

@Repository
class TrainRepository(private val mongoTemplate: MongoTemplate) {

    fun findAll(): List<TrainDocument> = mongoTemplate.findAll(TrainDocument::class.java)
}